openedx_tagging/core/tagging/models/base.py

```diff
@@ -170,6 +170,18 @@ def child_count(self) -> int:
             return self.taxonomy.tag_set.filter(parent=self).count()
         return 0
 
+    @cached_property
+    def descendant_count(self) -> int:
+        """
+        How many descendant tags this tag has in the taxonomy.
+        """
+        if self.taxonomy and not self.taxonomy.allow_free_text:
+            return self.taxonomy.tag_set.filter(
+                Q(parent__parent=self) |
+                Q(parent__parent__parent=self)
+            ).count() + self.child_count
```

Comment:
Can we combine these two calls together?

```suggestion
            return self.taxonomy.tag_set.filter(
                Q(parent=self) |
                Q(parent__parent=self) |
                Q(parent__parent__parent=self)
            ).count()
```

Reply:
We could, but if `child_count` is already cached (because it's a `@cached_property`), that does one set of extra database lookups that we don't need. I guess it's not a big deal if you think the consistency here is more important?

Reply:
Ah good point, and since we have to return the `child_count` too, it's better to leave this as-is. Suggestion retracted :)